File: 958 Check Completeness of a Binary Tree/958.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
from collections import deque
class Solution:
    def isCompleteTree(self, root: TreeNode) -> bool:
        if not root:
            return True
        flag = False
        queue = deque()
        queue.append(root)
        while queue:
            node = queue.popleft()
            if not node.left:
                flag = True
            elif flag:
                return False
            else:
                queue.append(node.left)
            
            if not node.right:
                flag = True
            elif flag:
                return False
            else:
                queue.append(node.right)
        return True

File: 958 Check Completeness of a Binary Tree/958.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isCompleteTree = function(root) {
    if (!root) {
        return true;
    }
    let flag = false;
    let queue = []
    queue.push(root)
    while (queue.length) {
        let node = queue.shift()
        if (!node.left) {
            flag = true
        } else if (flag) {
            return false;
        } else {
            queue.push(node.left)
        }

        if (!node.right) {
            flag = true
        } else if (flag) {
            return false;
        } else {
            queue.push(node.right)
        }
    }
    return true;
};